### PR TITLE
Support image dataset

### PR DESCRIPTION
**Pull request by zzachw**
- New dataset added
   - BaseImageDataset
   - SampleImageDataset

- New task added
   - covid19_classification_fn

- New model added
   - ResNet series

### CODE REVIEW

File: pyhealth/datasets/base_image_dataset.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    ):
+        # base attributes
+        self.dataset_name = (
+            self.__class__.__name__ if dataset_name is None else dataset_name
+        )
+        self.root = root
+        # TODO: dev seems unnecessary for image and signal?
+        self.dev = dev
+        if dev:
+            logger.warning("WARNING: dev has no effect for image datasets.")
```
Comment (ycq091044) [inline pyhealth/datasets/base_image_dataset.py:52]: why? cannot we load just a subset of images?

File: pyhealth/datasets/base_image_dataset.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        # TODO: refresh_cache seems unnecessary for image and signal?
+        self.refresh_cache = refresh_cache
+        if refresh_cache:
+            logger.warning("WARNING: refresh_cache has no effect for image datasets.")
+
+        self.metadata = pd.read_json(os.path.join(root, "metadata.jsonl"), lines=True)
```
Comment (ycq091044) [inline pyhealth/datasets/base_image_dataset.py:58]: Is it common to have .jsonl for every image dataset? Maybe move it to covid19_xray.py

File: pyhealth/models/resnet.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        self.cnn = models.__dict__[cnn_name](pretrained=pretrained)
+        hidden_dim = self.cnn.fc.in_features
+        self.label_tokenizer = self.get_label_tokenizer()
+        output_size = self.get_output_size(self.label_tokenizer)
+        self.cnn.fc = nn.Linear(hidden_dim, output_size)
+
+    def forward(self, **kwargs) -> Dict[str, torch.Tensor]:
+        """Forward propagation."""
+        # concat the info within one batch (batch, channel, length)
+        x = kwargs[self.feature_keys[0]]
```
Comment (ycq091044) [inline pyhealth/models/resnet.py:55]: we should constrain the size of feature_keys is 1.

**Comment by ycq091044** [comment]
also, add your new dataset and model to docs/ and logs.rst

**Comment by zzachw** [comment]
Close this PR for now. Will submit a new PR for image support this week.

**Pull request closed without merging**